code/game/objects/structures/barsign.dm

```diff
@@ -1,34 +1,46 @@
-/// All Signs are 64 by 32 pixels, they take two tiles
-/obj/structure/sign/barsign
+#define BARSIGN_FRAME		0
+#define BARSIGN_CIRCUIT		1
+#define BARSIGN_WIRED		2
+#define BARSIGN_COMPLETE	3
+
+
+
+/obj/machinery/barsign // All Signs are 64 by 32 pixels, they take two tiles
 	name = "Bar Sign"
 	desc = "A bar sign with no writing on it."
 	icon = 'icons/obj/barsigns.dmi'
-	icon_state = "empty"
+	icon_state = "off"
 	req_access = list(ACCESS_BAR)
-	max_integrity = 500
-	integrity_failure = 250
-	armor = list(MELEE = 20, BULLET = 20, LASER = 20, ENERGY = 100, BOMB = 0, RAD = 0, FIRE = 50, ACID = 50)
-	var/list/barsigns=list()
-	var/list/hiddensigns
+	max_integrity = 200
+	integrity_failure = 160
+	idle_power_consumption = 2
+	active_power_consumption = 4
```

Comment:
How much it was before? It could be higher than this

Reply:
I don't think it drains any power at the moment, but I will increase it.